File: holic/startng/src/app/model/principal.ts
export class Principal {
  id: number;
  name: string;
  firstName: string;
  email: string;
  login: string;
  password: string;
  country: string;
  age: number;
  admin: boolean;

  static initPrincipal(principal: Principal): Principal {
    const p = new Principal();
    p.id = principal.id;
    p.name = principal.name;
    p.firstName = principal.firstName;
    p.email = principal.email;
    p.login = principal.login;
    p.password = principal.password;
    p.country = principal.country;
    p.age = principal.age;
    p.admin = principal.admin;

    return p;
  }
}

export class RegisterView {
  name: string;
  //firstName: string;
  email: string;
  login: string;
  password: string;
  //country: string;
  //age: number;
  //admin: boolean;
}

File: holic/src/main/java/com/video/holic/model/user/Principal.java
package com.video.holic.model.user;

import lombok.Builder;
import lombok.Data;

import java.io.Serializable;

@Data
@Builder
public class Principal implements Serializable {
    private Integer id;
    private String name;
    private String firstName;
    private String email;
    private String login;
    private String password;
    private String country;
    private short age;
    private boolean admin;
}
